### PR TITLE
refactor: init input validations

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

Users may not have linting enabled on their IDEs, init type validations help discover issues early on (on the app startup)

## Test Plan
Unit tests for all validations and errors

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] ~~`coreDriverInterfaceSupported.json` file has been updated (if needed)~~
    -   ~~Along with the associated array in `supertokens_python/constants.py`~~
-   [ ] ~~`frontendDriverInterfaceSupported.json` file has been updated (if needed)~~
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] ~~If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable~~
-   [ ] ~~If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py~~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
 
+## [0.8.2]
```
Comment (rishabhpoddar) [inline CHANGELOG.md:9]: I am not going to be releasing this change now, so can you put all of this in the unreleased section? And revert the changes related to version change in this PR

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (sattvikc) [reply CHANGELOG.md:9]: okay

**Review by rishabhpoddar — APPROVED** [review]
